header_enum_explicit_type_hpp test fails on aarch64

The error is:
```
rust-bindgen/tests/headers/enum_explicit_type.hpp:9:16: error: enumerator value evaluates to -1, which cannot be narrowed to type 'char' [-Wc++11-narrowing], err: true
```

I might be wrong but I think the problem is that `char` in c++ (when neither signed nor unsigned) is implementation defined.
In x86 `char` is defined as signed and in aarch64 it is unsigned, as can be seen by the fact that in the `libc` crate, `c_char` is an alias to `i8` and `u8` in each platform respectively. 


The flag are default when running `cargo test`

Device: odroid C2

Linux kernel: `3.14.79-27-ARCH`

Clang: `4.0.1`

GCC: `7.1.1 20170516`

Host: `nightly-aarch64-unknown-linux-gnu`

Rustc: `1.21.0-nightly (aac223f4f 2017-07-30)`

--- a/tests/headers/enum_explicit_type.hpp
+++ b/tests/headers/enum_explicit_type.hpp
@@ -5,7 +5,7 @@ enum Foo: unsigned char {
     Qux
 };
 
-enum Neg: char {
+enum Neg: signed char {
     MinusOne = -1,
     One = 1,
 };

